src/osip-simulation-view/src/main/java/edu/kit/pse/osip/simulation/view/settings/SimulationSettingsWindow.java

```diff
@@ -113,6 +117,28 @@ private BorderPane createRootLayout() {
     }
 
     /**
+     * Updates enabled state of the save button
+     */
+    private void updateSaveButton() {
+        boolean allValid = true;
+        for (Map.Entry<TankSelector, PortTextField> entry : portTextFields.entrySet()) {
+            if (!FormatChecker.isValidPort(entry.getValue().getText())) {
+                allValid = false;
+            }
+        }
+        btnSave.setDisable(!allValid);
+    }
+
+    /**
+     * Updates the settings file with the new ports
+     */
+    private void updateSettingsFile() {
+        for (Map.Entry<TankSelector, PortTextField> entry : portTextFields.entrySet()) {
+            settings.setServerPort(entry.getKey(), Integer.parseInt(entry.getValue().getText()));
```

Comment:
entry.getValue().getPort() liefert doch schon den Port, oder? Damit ist das Parsen in den Integer hier unnötig. Außerdem überschneidet sich das hier bereits mit Davids Implementierung des SimulationControllers, wenn ich richtig gesehen habe.